ansible/configs/sap-hana/pre_infra.yml

```diff
@@ -28,3 +28,13 @@
       args:
         creates: "{{output_dir}}/{{env_authorized_key}}.pub"
       when: set_env_authorized_key | bool
+
+    - name: Find out SnapShot ID for SAP Software
+      ec2_snapshot_info:
+        filters:
+          description: "{{ sap_software_snap_name }}"
+      register: sap_snap
+
+    - name: Set local fact for sap_software_snap_id
+      set_fact:
+        sap_software_snap_id: "{{ sap_snap.snapshots[0].snapshot_id }}"
```

Comment:
Did you test when the snapshot is not found ?
What should be our strategy if the snapshot is not found ?

Reply:
The deployment must fail then, will capture that